src/components/Layout/Layout.tsx

```diff
@@ -0,0 +1,75 @@
+import { ReactNode } from 'react'
+import { cx } from 'class-variance-authority'
+
+import { findElement } from '@/utils/findElement'
+
+export interface LayoutProps {
+  children: ReactNode
+  className?: string
+}
+
+export const Layout = ({ children, className, ...props }: LayoutProps) => {
+  const findLayoutElement = findElement(children)
+
+  const topBanner = findLayoutElement('Layout.TopBanner')
+  const header = findLayoutElement('Layout.Header')
+  const hero = findLayoutElement('Layout.Hero')
+  const leadingPanel = findLayoutElement('Layout.LeadingPanel')
+  const leadingSidebar = findLayoutElement('Layout.LeadingSidebar')
+  const content = findLayoutElement('Layout.Content')
+  const trailingSidebar = findLayoutElement('Layout.TrailingSidebar')
+  const trailingPanel = findLayoutElement('Layout.TrailingPanel')
+  const bottomBanner = findLayoutElement('Layout.BottomBanner')
+  const footer = findLayoutElement('Layout.Footer')
```

Comment:
Is this truly needed ? I think we could just use the component

Reply:
Yes, but we are gonna have to maintain the layout on every url.

At that moment we have bugs with:
- layout shifting between different doc pages (and it will be harder when we start scaling it maintainig it 1 by 1)
- nav and displaying/hidding section changes per breakpoint might be handled once not depending-coupling on what element is placed inside.
- When we want to create layout changes we will need to fix it on every element or just once in the layout

Reply:
Sorry but I don't get how these bugs are related with the proposed implementation

Reply:
the current header 
<img width="1552" alt="image" src="https://github.com/adevinta/spark-website/assets/1674036/c4342f83-eaa1-488d-a32c-8fb38b97fec6">
PR layout
<img width="1769" alt="image" src="https://github.com/adevinta/spark-website/assets/1674036/75a9d434-e855-4336-865b-47536fde1f58">


the designed layout
<img width="1271" alt="image" src="https://github.com/adevinta/spark-website/assets/1674036/d9a720d6-eb94-49f4-9f2f-1aa9a166a1b0">

Reply:
yes, 1 by 1 when scaling will be a jigsaw headache

Reply:
According to what I see it was following the specs